python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_graph.py

```diff
@@ -218,3 +227,61 @@ def child(parent):
     assert external_asset_graph.get_parents_partitions(child.key, "2") == {
         AssetKeyPartitionKey(parent.key, "2")
     }
+
+
+def test_required_multi_asset_sets_non_subsettable_multi_asset():
+    @multi_asset(outs={"a": AssetOut(dagster_type=None), "b": AssetOut(dagster_type=None)})
+    def non_subsettable_multi_asset():
+        ...
+
+    for asset_graph in [
+        AssetGraph.from_assets([non_subsettable_multi_asset]),
+        to_external_asset_graph([non_subsettable_multi_asset]),
+    ]:
+        for asset_key in non_subsettable_multi_asset.keys:
+            assert (
+                asset_graph.get_required_multi_asset_keys(asset_key)
+                == non_subsettable_multi_asset.keys
+            )
+
+
+def test_required_multi_asset_sets_subsettable_multi_asset():
+    @multi_asset(
+        outs={"a": AssetOut(dagster_type=None), "b": AssetOut(dagster_type=None)}, can_subset=True
+    )
+    def subsettable_multi_asset():
+        ...
+
+    for asset_graph in [
+        AssetGraph.from_assets([subsettable_multi_asset]),
+        to_external_asset_graph([subsettable_multi_asset]),
+    ]:
+        for asset_key in subsettable_multi_asset.keys:
+            assert asset_graph.get_required_multi_asset_keys(asset_key) == set()
+
+
+def test_required_multi_asset_sets_graph_backed_multi_asset():
+    @op
+    def op1():
+        return 1
+
+    @op(out={"b": Out(), "c": Out()})
+    def op2():
+        return 4, 5
+
+    @graph(out={"a": GraphOut(), "b": GraphOut(), "c": GraphOut()})
+    def graph1():
+        a = op1()
+        b, c = op2()
+        return a, b, c
+
+    graph_backed_multi_asset = AssetsDefinition.from_graph(graph1)
```

Comment:
I think this logic might break down if you have two graph backed multi assets that derive from the same graph. It may all work out, but should be explicitly tested here.

In general, I'm not sure how much I trust those name fields to serve as a unique identifier for a node. You may get weird cases when you have multiple partitioned __ASSET_JOBS, which each reference copies of the same asset graph, and so therefore get the same graph_name or something like that.

An alternative, much safer, solution would be to include a "required_multi_asset_keys" field on the `ExternalAssetNode`. However, this is pretty inefficient storage-wise. An iteration on that would be to put an `md5hash(frozenset(required_multi_asset_keys))` as the unique identifier.